Node/core-DirectLineWebSockets/DirectLineClient/app.js

```diff
@@ -0,0 +1,195 @@
+        return client;
+    })
+    .catch(function (err) {
+        console.error('Error initializing DirectLine client', err);
+    });
+
+// Once the client is ready, create a new conversation 
+directLineClient.then(function (client) {
+    client.Conversations.Conversations_StartConversation()
+        .then(function (response) {
+            var responseObj = response.obj;
+
+            // Start console input loop from stdin
+            sendMessagesFromConsole(client, responseObj.conversationId);
+            
+            if (useW3CWebSocket) {
+                // Start receiving messages from WS stream - using W3C client
+                startReceivingW3CWebSocketClient(responseObj.streamUrl, responseObj.conversationId);
+            } else {
+                // Start receiving messages from WS stream - using Node client
+                startReceivingWebSocketClient(responseObj.streamUrl, responseObj.conversationId);
+            }
+        });
+});
+
+// Read from console (stdin) and send input to conversation using DirectLine client
+function sendMessagesFromConsole(client, conversationId) {
+    var stdin = process.openStdin();
+    process.stdout.write('Command> ');
+    stdin.addListener('data', function (e) {
+        var input = e.toString().trim();
+        if (input) {
+            if (input.toLowerCase() === 'exit') {
+                return process.exit();
+            }
+
+            // Send message
+            client.Conversations.Conversations_PostActivity(
+                {
+                    conversationId: conversationId,
+                    activity: {
+                        textFormat: 'plain',
+                        text: input,
+                        type: 'message',
+                        from: {
+                            id: directLineClientName,
+                            name: directLineClientName
+                        }
+                    }
+                }).catch(function (err) {
+                    console.error('Error sending message:', err);
+                });
+
+            process.stdout.write('Command> ');
+        }
+    });
+}
+
+function startReceivingWebSocketClient(streamUrl, conversationId) {
+    console.log('Starting WebSocket Client for message streaming on conversationId: ' + conversationId);
+
+    var ws = new (require('websocket').client)();
+
+    ws.on('connectFailed', function(error) {
+        console.log('Connect Error: ' + error.toString());
+    });
+     
+    ws.on('connect', function(connection) {
+        console.log('WebSocket Client Connected');
+        connection.on('error', function(error) {
+            console.log("Connection Error: " + error.toString());
+        });
+        connection.on('close', function() {
+            console.log('WebSocket Client Disconnected');
+        });
+        connection.on('message', function(message) {
+            if (message.type === 'utf8' && message.utf8Data.length > 0) {
```

Comment:
Comment similar to:

// Occasionally, the Direct Line service sends an empty message as a liveness ping. Ignore these messages.